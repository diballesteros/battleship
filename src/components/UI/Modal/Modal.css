.modal-background {
    position: absolute;
    top: 0;
    right: 0;
    height: 100%;
    width: 100%;
    z-index: 1;
    background:  rgba(0,0,0,0.5);
}

.modal {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: space-evenly;
    position: absolute;
    top: 30%;
    right: 46%;
    height: 200px;
    width: 200px;
    background: white;
    border: 2px solid black;
    border-radius: 4px;
    transition: 1.1s ease-out;
    filter: blur(0);
    transform: scale(1);
    opacity: 1;
}