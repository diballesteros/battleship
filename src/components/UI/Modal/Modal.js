import React from 'react';
import './Modal.css';
import Button from '../button/Button';

const Modal = (props) => {
    return (
        <div className="modal-background">
            <div className="modal">
                <label>{props.modalText}</label>
                <Button clicked={() => props.modalFn()}>{props.children}</Button>
            </div> 
        </div> 
    );
}

export default Modal;