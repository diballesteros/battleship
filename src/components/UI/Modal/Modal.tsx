import React from 'react';
import './Modal.css';
import Button from '../button/Button';

interface ModalProps {
  children: string;
  modalFn: () => void;
  modalText: string;
}

const Modal: React.FC<ModalProps> = ({ children, modalText, modalFn }) => (
  <div className="modal-background">
    <div className="modal">
      <h2>{modalText}</h2>
      <Button clicked={() => modalFn()} disabled={false}>
        {children}
      </Button>
    </div>
  </div>
);

export default Modal;
