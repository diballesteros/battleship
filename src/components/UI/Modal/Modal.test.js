import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Modal from './Modal';

describe('Modal component', () => {
    test('Modal initalized', () => {
        const { asFragment } = render(<Modal modalText={'Game over'}>New Game</Modal>);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Modal button click', () => {
        const mockFn = jest.fn();
        const { container } = render(<Modal modalText={'Game over'} modalFn={() => mockFn()}>New Game</Modal>);
        const component = container.querySelector('button');
        fireEvent.click(component);
        expect(mockFn).toHaveBeenCalled();
    });
});
