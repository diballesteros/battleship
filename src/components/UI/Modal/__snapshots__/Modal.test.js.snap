// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Modal initalized 1`] = `
<DocumentFragment>
  <div
    class="modal-background"
  >
    <div
      class="modal"
    >
      <label>
        Game over
      </label>
      <button>
        New Game
      </button>
    </div>
  </div>
</DocumentFragment>
`;
